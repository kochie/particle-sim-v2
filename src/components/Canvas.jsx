import React from 'react';
import init from '../lib/init';

export default class Canvas extends React.Component {
  componentDidMount() {
    this.renderCanvas(init(this.canvasElement));
  }

  renderCanvas(env) {
    window.requestAnimationFrame(() => {
      this.renderCanvas(env);
    });
    // console.log(env.camera.getWorldDirection());
    // TWEEN.update();

    env.stats.begin();
    env.controls.update();
    env.renderer.render(env.scene, env.camera);
    env.stats.end();
  }

  render() {
    return (
      <div ref={(element) => { this.canvasElement = element; }} />
    );
  }
}
