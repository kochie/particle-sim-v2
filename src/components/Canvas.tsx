import * as React from 'react';

import init from '../lib/init';
import Environment from '../lib/environment';

export default class Canvas extends React.Component {
  public canvasElement: HTMLCanvasElement

  public componentDidMount(): void {
    this.renderCanvas(init(this.canvasElement));
  }

  private renderCanvas(env: Environment): void {
    window.requestAnimationFrame(() => {
      this.renderCanvas(env);
    });

    env.stats.begin();
    env.controls.update();
    env.renderer.render(env.scene, env.camera);
    env.stats.end();
  }

  public render(): React.ReactElement {
    return (
      <canvas ref={(element) => { this.canvasElement = element }} />
    );
  }
}
