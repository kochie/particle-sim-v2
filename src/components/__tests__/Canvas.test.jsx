import renderer from 'react-test-renderer';
import React from 'react';

import Canvas from '../Canvas';


jest.mock('../../lib/init', () => jest.fn(() => ({
  stats: {
    begin: jest.fn(),
    end: jest.fn(),
  },
  controls: {
    update: jest.fn(),
  },
  renderer: {
    render: jest.fn(),
  },
})));

function init(props) {
  const instance = renderer.create(<Canvas {...props} />);
  return { instance };
}

describe('canvas react component', () => {
  it('should render and match snapshot', () => {
    const { instance } = init();
    expect(instance).toMatchSnapshot();
  });
});
