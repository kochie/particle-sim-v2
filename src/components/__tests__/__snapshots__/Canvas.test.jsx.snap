// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`canvas react component should render and match snapshot 1`] = `<div />`;
